dss/events/handlers/index.py

```diff
@@ -24,47 +24,60 @@
 DSS_BUNDLE_KEY_REGEX = r"^bundles/[0-9A-Fa-f]{8}-[0-9A-Fa-f]{4}-4[0-9A-Fa-f]{3}-[0-9A-Fa-f]{4}-[0-9A-Fa-f]{12}\..+$"
 
 
-def process_new_s3_indexable_object(event, logger) -> None:
-    try:
-        # This function is only called for S3 creation events
-        key = unquote(event['Records'][0]["s3"]["object"]["key"])
-        bucket_name = event['Records'][0]["s3"]["bucket"]["name"]
-        process_new_indexable_object(bucket_name, key, Replica.aws, logger)
-    except Exception as ex:
-        logger.error("Exception occurred while processing S3 event: %s Event: %s", ex, json.dumps(event, indent=4))
-        raise
-
-
-def process_new_gs_indexable_object(event, logger) -> None:
-    try:
-        # This function is only called for GS creation events
-        bucket_name = event["bucket"]
-        key = event["name"]
-        process_new_indexable_object(bucket_name, key, Replica.gcp, logger)
-    except Exception as ex:
-        logger.error("Exception occurred while processing GS event: %s Event: %s", ex, json.dumps(event, indent=4))
-        raise
-
-
-def process_new_indexable_object(bucket_name: str, key: str, replica: Replica, logger) -> None:
-    if is_bundle_to_index(key):
-        logger.info(f"Received {replica} creation event for bundle which will be indexed: {key}")
-        document = BundleDocument.from_bucket(replica, bucket_name, key, logger)
-        index_name = document.prepare_index()
-        document.add_to_index(index_name)
-        document.notify_matching_subscribers(index_name)
-        logger.debug(f"Finished index processing of {replica} creation event for bundle: {key}")
-    else:
-        logger.debug(f"Not indexing {replica} creation event for key: {key}")
+class IndexHandler:
+
+    @classmethod
+    def process_new_indexable_object(cls, event, logger) -> None:
+        raise NotImplementedError("`process_new_indexable_object` is not implemented!")
```

Comment:
I've seen single quotes being used for nested quotations this across the project.